dubbo-remoting/dubbo-remoting-etcd3/src/main/java/org/apache/dubbo/remoting/etcd/jetcd/JEtcdClient.java

```diff
@@ -447,7 +447,11 @@ protected void closeWatchRequest() {
             if (this.watchRequest == null) {
                 return;
             }
-            this.watchRequest.onCompleted();
```

Comment:
change to :

```java
protected void closeWatchRequest() {
    if (this.watchRequest == null) {
        return;
    }

    try{
        WatchCancelRequest watchCancelRequest =
                WatchCancelRequest.newBuilder().setWatchId(watchId).build();
        WatchRequest cancelRequest = WatchRequest.newBuilder()
                .setCancelRequest(watchCancelRequest).build();
        watchRequest.onNext(cancelRequest);
    }finally {
        this.watchRequest.onCompleted();
        this.watchRequest = null;
    }

}
```

Reply:
I think it's fine. We can always reformat the code at any time, considering the author fails to response. I will merge it.